src/Fetcher/ConfTechFetcher.php

```diff
@@ -13,117 +13,90 @@
+    private const SOURCE = 'conf-tech';
+
+    // These are all the tags that are available for fetching from confs.tech.
+    // If you feel like one is missing, feel free to add one from https://github.com/tech-conferences/conference-data/tree/main/conferences
+    private const SOURCE_AVAILABLE_TAGS = [
         'android',
+        'clojure',
+        'cpp',
         'css',
         'data',
         'devops',
         'dotnet',
         'elixir',
+        'elm',
         'general',
         'golang',
         'graphql',
         'ios',
+        'iot',
+        'java',
         'javascript',
+        'kotlin',
+        'leadership',
+        'networking',
         'php',
+        'product',
         'python',
         'ruby',
         'rust',
+        'scala',
         'security',
         'tech-comm',
+        'typescript',
         'ux',
     ];
 
-    private $em;
-    private $conferenceRepository;
-    private $httpClient;
-    private $serializer;
-    private $logger;
-    private $tagRepository;
-    private $locationGuesser;
-    private $slugger;
-
-    public function __construct(
-        ManagerRegistry $doctrine,
-        SerializerInterface $serializer,
-        LoggerInterface $logger,
-        LocationGuesser $locationGuesser
-    ) {
-        $this->em = $doctrine->getManager();
-        // @todo replace with proper DI when http-client will be released as stable
-        $this->httpClient = HttpClient::create();
-        $this->serializer = $serializer;
-        $this->logger = $logger;
-        $this->conferenceRepository = $this->em->getRepository(Conference::class);
-        $this->tagRepository = $this->em->getRepository(Tag::class);
-        $this->locationGuesser = $locationGuesser;
-        $this->slugger = new AsciiSlugger();
-    }
-
-    public function isActive(): bool
-    {
-        return true;
-    }
+    private LocationGuesser $locationGuesser;
+    private HttpClientInterface $client;
+    private LoggerInterface $logger;
 
-    public function getUrl(array $params = []): string
+    public function __construct(LocationGuesser $locationGuesser, ?HttpClientInterface $client = null, ?LoggerInterface $logger = null)
     {
-        return "https://raw.githubusercontent.com/tech-conferences/conference-data/master/conferences/$params[date]/$params[tag].json";
+        $this->locationGuesser = $locationGuesser;
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger ?: new NullLogger();
     }
 
-    public function fetch(): \Generator
+    public function fetch(array $configuration = []): ?\Generator
     {
         foreach ([date('Y'), date('Y', strtotime('+1 year'))] as $date) {
-            foreach ($this->tagRepository->findBy(['selected' => true]) as $tag) {
-                $tagKey = array_search($this->slugger->slug(strtolower($tag->getName()))->toString(), self::TAGS_SYNONYMS);
+            if (0 === \count($configuration) || 0 === \count($configuration['tags'])) {
```

Comment:
Shouldn't we pass the FetcherConfiguration object here (which will act as a Value Object). May be with a dedicated method in the FetcherConfiguration to check if it's ok will be relevant? Something like `$fetcherConfig->isComplete()`?

Reply:
Not sure it's useful. **and** it's not possible to add custom method on $fetcher config since it's generic for all fetcher